src/go/configgenerator/cluster_generator_test.go

```diff
@@ -579,6 +579,134 @@ func TestMakeBackendRoutingCluster(t *testing.T) {
 	}
 }
 
+func TestMakeCatchAllBackendClusterCluster(t *testing.T) {
+	testData := []struct {
+		desc               string
+		fakeServiceConfig  *confpb.Service
+		BackendAddress     string
+		DnsResolverAddress string
+		tlsContextSni      string
+		wantedCluster      *v2pb.Cluster
+		wantedError        string
+	}{
+		{
+			desc: "test DnsResolverAddress in form of IP_ADDR:PORT",
+			fakeServiceConfig: &confpb.Service{
+				Name: testProjectName,
+				Apis: []*apipb.Api{
+					{
+						Name: "1.cloudesf_testing_cloud_goog",
+						Methods: []*apipb.Method{
+							{
+								Name: "Foo",
+							},
+						},
+					},
+				},
+			},
+			BackendAddress:     "http://127.0.0.1:80",
+			DnsResolverAddress: "127.0.0.1:53",
+			wantedCluster: &v2pb.Cluster{
+				Name:                 "bookstore.endpoints.project123.cloud.goog_local",
+				ConnectTimeout:       ptypes.DurationProto(20 * time.Second),
+				ClusterDiscoveryType: &v2pb.Cluster_Type{v2pb.Cluster_LOGICAL_DNS},
+				LoadAssignment:       util.CreateLoadAssignment("127.0.0.1", 80),
```

Comment:
This cluster config looks weird,  cluster address is already IP,  it doesn't need DNS resolver. 

maybe we can put in a check,  if host is ready IP,  not need to add DNS resolver.

Reply:
Why not leave envoy to handle IP address by itself?